cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java

```
@@ -242,7 +250,79 @@ public void updateAssignmentGrade() throws Exception {
 		updatedag.setScore("88");
 		verify(assignmentGradeRepository, times(1)).save(updatedag);
 	}
+	
+	
+	@Test
+	@DisplayName("Test addAssignment")
+	public void testAddAssignment() throws Exception {
```

The issue with writing a long test like this that checks multiple behaviors is if it fails early, anything later on in the test is unreachable and therefore some behaviors go untested.  I'd consider refactoring this into several smaller unit tests.